dex-core/src/ErgoDex/Amm/PoolActions.hs

```diff
@@ -122,7 +122,28 @@ runDeposit' executorPkh (Confirmed depositOut Deposit{..}) (poolOut, pool@Pool{.
       | isAda poolCoinY = (inX, inY - retagAmount exFee - retagAmount adaCollateral)
       | otherwise       = (inX, inY)
 
-    pp@(Predicted nextPoolOut _) = applyDeposit pool (netInX, netInY)
+    lqOutput = liquidityAmount pool (netInX, netInY)
+    mintLqValue = assetAmountValue lqOutput
+
+    lq = unAmount poolLiquidity
+    poolX  = unAmount poolReservesX
+    poolY  = unAmount poolReservesY
+    minByX = (unAmount netInX) * lq `div` poolX
+    minByY = (unAmount netInY) * lq `div` poolY
+
+    (changeX, changeY) =
+      if (minByX == minByY)
+      then (0, 0)
+      else
+        if (minByX < minByY)
+        then (0, (minByY - minByX) * poolY `div` lq)
+        else ((minByX - minByY) * poolX `div` lq, 0)
+
+    alignmentValue =
+         assetClassValue (unCoin poolCoinY) changeY
+      <> assetClassValue (unCoin poolCoinY) changeX
+
+    pp@(Predicted nextPoolOut _) = applyDeposit pool (netInX - (Amount changeX), netInY - (Amount changeY))
```

Comment:
Why not making calculations in applyDeposit? All the math reside there, PoolActions is about transaction candidate assembly.